src/core/src/graph_util.cpp

```diff
@@ -245,6 +245,12 @@ std::vector<std::shared_ptr<ngraph::Node>> ngraph::clone_nodes(const std::vector
                 new_output.get_rt_info() = output_rt_info;
             }
 
+            for (auto input : node->inputs()) {
+                const auto& output_rt_info = input.get_rt_info();
+                auto new_input = input.for_node(cloned_node);
```

Comment:
From my point of view `for_node` is not required as we can use `cloned_node->input(input.get_index())`

Reply:
Ok! changed for_node to cloned_node->input(input.get_index())